modules/vcf2maf/1.3/vcf2maf.smk

```diff
@@ -137,66 +161,162 @@ rule _vcf2maf_gnomad_filter_maf:
+    wildcard_constraints:
+        target_build = "|".join(CFG["options"]["target_builds"])
     run:
         op.relative_symlink(input.maf, output.maf)
-        op.relative_symlink((input.maf_converted+str("_")+str(params.chain)+str(".maf")), (output.maf[:-4]+str(".converted_")+str(params.chain)+str(".maf")))
+
+rule _vcf2maf_output_maf_native:
+    input:
+        maf = str(rules._vcf2maf_normalize_prefix.output.maf),
+    output:
+        maf = CFG["dirs"]["outputs"] + "{seq_type}--{target_build}/{tumour_id}--{normal_id}--{pair_status}_{base_name}.maf"
+    wildcard_constraints:
+        target_build = "|".join(CFG["options"]["target_builds"])
+    run:
+        op.relative_symlink(input.maf, output.maf)
+
+
+def specify_output_folder(wildcards):
+    original_genome_build = get_original_genome(wildcards)
+    target_genome_build = wildcards.target_build
+    # Sanity check that this genome build is specified in the reference config provided
+    if not target_genome_build in VCF2MAF_GENOME_VERSION:
+        raise AttributeError(f"Target genome build {target_genome_build} is not specified in reference config {VCF2MAF_REFERENCE_CONFIG}")
+
+    # If this MAF was converted from a different genome build, specify a different output filter
+    if VCF2MAF_GENOME_VERSION[original_genome_build] != VCF2MAF_GENOME_VERSION[target_genome_build]:
+        return rules._vcf2maf_output_maf_projection.output.maf
+    else:
+        return rules._vcf2maf_output_maf_native.output.maf
+
+rule _vcf2maf_output_dispatch:
+    input:
+        maf = specify_output_folder
+    output:
+        dispatch = CFG["dirs"]["_parent"] + "dispatch/{seq_type}--{target_build}/{tumour_id}--{normal_id}--{pair_status}_{base_name}.dispatch"
+    shell:
+        "touch {output.dispatch}"
 
 # Generates the target sentinels for each run, which generate the symlinks
 rule _vcf2maf_all:
     input:
-        expand(str(rules._vcf2maf_output_maf.output.maf), zip,
-            seq_type = CFG["runs"]["tumour_seq_type"],
-            genome_build = CFG["runs"]["tumour_genome_build"],
-            tumour_id = CFG["runs"]["tumour_sample_id"],
-            normal_id = CFG["runs"]["normal_sample_id"],
-            pair_status = CFG["runs"]["pair_status"],
-            base_name = [CFG["vcf_base_name"]] * len(CFG["runs"]["tumour_sample_id"]))
+        expand(
+            expand(str(rules._vcf2maf_output_dispatch.output.dispatch), zip,
+                seq_type = CFG["runs"]["tumour_seq_type"],
+                tumour_id = CFG["runs"]["tumour_sample_id"],
+                normal_id = CFG["runs"]["normal_sample_id"],
+                pair_status = CFG["runs"]["pair_status"],
+                base_name = [CFG["vcf_base_name"]] * len(CFG["runs"]["tumour_sample_id"]),
+                allow_missing = True),
+            target_build = CFG["options"]["target_builds"])
```

Comment:
Can you please show an example of what does the output path for each genome build looks like now?

Reply:
99 outputs will generate the following subfolders:
`<seq_type>--<genome_build>
<seq_type>--<genome_build>--projection`
For instance, if a VCF was originally referenced against hg38, the hg38 and grch38 MAFs will be placed in `<seq_type>--<genome_build>`, while the hg19 and grch37 MAFs will be placed in `<seq_type>--<genome_build>--projection`

Reply:
We need them all in the SAME directory per projection regardless of their parental genome_build

Reply:
Done and implemented as of current commit